File: Semestrul 4/SDI/lab09-10/server .net/TriathlonModel/Discipline.cs
using System;

namespace Triathlon.Model
{
    public enum Discipline
    {
        RUNNING = 1,
        SWIMMING = 2,
        CYCLING = 3
    }

    public static class DisciplineExtensions
    {
        public static long GetId(this Discipline discipline)
        {
            return (long)discipline;
        }

        public static Discipline GetById(long id)
        {
            foreach (Discipline discipline in Enum.GetValues(typeof(Discipline)))
            {
                if ((long)discipline == id)
                {
                    return discipline;
                }
            }
            throw new ArgumentException($"No Discipline with id {id}");
        }
    }
}

File: Semestrul 4/SDI/lab11/java/TriathlonModel/src/main/java/triathlon/model/Discipline.java
package triathlon.model;

import java.util.Objects;

public enum Discipline {
    RUNNING(1L),
    SWIMMING(2L),
    CYCLING(3L);

    private final Long id;

    Discipline(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }
    public static Discipline getById(Long id) {
        for (Discipline discipline : values()) {
            if (Objects.equals(discipline.getId(), id)) {
                return discipline;
            }
        }
        throw new IllegalArgumentException("No Discipline with id " + id);
    }
}
